// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kvstore;
service KVStore {
  rpc read (ReadRequest) returns (ReadReply) {}
  rpc write (WriteRequest) returns (WriteReply) {}
  rpc getServerTag (TagRequest) returns (TagReply) {}
}

message ReadRequest {
  uint32 client_id = 1;
  string key = 2;
}

message ReadReply {
  uint32 timestamp = 1;
  string value = 2;
}

message WriteRequest {
  uint32 client_id = 1;
  uint32 timestamp = 2;
  string key = 3;
  string value = 4;
}

message WriteReply {
  uint32 ack = 1;
}

message TagRequest {
  uint32 client_id = 1;
}

message TagReply {
  uint32 tag = 1;
}